ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -855,13 +855,35 @@ mbfl_strpos(
 		needle_u8 = needle;
 	}
 
-	if (needle_u8->len < 1) {
-		result = (size_t) -8;
+	result = (size_t) -1;
+	if (haystack_u8->len < needle_u8->len) {
 		goto out;
 	}
 
-	result = (size_t) -1;
-	if (haystack_u8->len < needle_u8->len) {
+	if (needle_u8->len == 0) {
+		size_t haystack_length = mbfl_strlen(haystack_u8);
+		/* Check if offset is out of bound */
+		if (
+			(offset > 0 && offset > haystack_length)
+			|| (offset < 0 && -offset > haystack_length)
+		) {
+			result = -16;
+			goto out;
+		}
+
+		if (reverse) {
+			if (offset < 0) {
+				result = haystack_length + offset;
+			} else {
+				result = haystack_length;
+			}
+		} else {
+			if (offset < 0) {
+				result = haystack_length + offset;
+			} else {
```

Comment:
Should I move the ``if (offset < 0)`` condition out? As it's the same for both now.